pkg/kubelet/kubelet_server_journal.go

```diff
@@ -0,0 +1,431 @@
+	}
+
+	var sinceTime time.Time
+	sinceTimeValue := query.Get("sinceTime")
+	if len(sinceTimeValue) > 0 {
+		sinceTime, err = time.Parse(time.RFC3339, sinceTimeValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("sinceTime"), sinceTimeValue, "invalid time format"))
+		} else {
+			nlq.SinceTime = &sinceTime
+		}
+	}
+
+	var untilTime time.Time
+	untilTimeValue := query.Get("untilTime")
+	if len(untilTimeValue) > 0 {
+		untilTime, err = time.Parse(time.RFC3339, untilTimeValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("untilTime"), untilTimeValue, "invalid time format"))
+		} else {
+			nlq.UntilTime = &untilTime
+		}
+	}
+
+	var boot int
+	bootValue := query.Get("boot")
+	if len(bootValue) > 0 {
+		boot, err = strconv.Atoi(bootValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("boot"), bootValue, err.Error()))
+		} else {
+			nlq.Boot = &boot
+		}
+	}
+
+	var tailLines int
+	tailLinesValue := query.Get("tailLines")
+	if len(tailLinesValue) > 0 {
+		tailLines, err = strconv.Atoi(tailLinesValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("tailLines"), tailLinesValue, err.Error()))
+		} else {
+			nlq.TailLines = &tailLines
+		}
+	}
+
+	pattern := query.Get("pattern")
+	if len(pattern) > 0 {
+		nlq.Pattern = pattern
+	}
+
+	if len(allErrs) > 0 {
+		return nil, allErrs
+	}
+
+	if reflect.DeepEqual(nlq, nodeLogQuery{}) {
+		return nil, allErrs
+	}
+
+	return &nlq, allErrs
+}
+
+func validateServices(services []string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	var totalLength int
+	for _, s := range services {
+		totalLength += len(s)
+		if err := safeString(s); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("query"), s, err.Error()))
+		}
+	}
+
+	maxTotalServiceNamesLength := 1000
+	if totalLength > maxTotalServiceNamesLength {
+		allErrs = append(allErrs, field.TooLongMaxLength(field.NewPath("query"), services, maxTotalServiceNamesLength))
+	}
+	return allErrs
+}
+
+func (n *nodeLogQuery) validate() field.ErrorList {
+	allErrs := validateServices(n.Services)
+	switch {
+	case len(n.Files) == 0 && len(n.Services) == 0:
+		allErrs = append(allErrs, field.Required(field.NewPath("query"), "cannot be empty with options"))
+	case len(n.Files) > 0 && len(n.Services) > 0:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), fmt.Sprintf("%v, %v", n.Files, n.Services),
+			"cannot specify a file and service"))
+	case len(n.Files) > 1:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), n.Files, "cannot specify more than one file"))
+	case len(n.Files) == 1 && n.options != (options{}):
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), n.Files, "cannot specify file with options"))
+	case len(n.Files) == 1:
+		if _, err := os.Stat(filepath.Join(logDir, n.Files[0])); err != nil {
```

Comment:
Can use something like https://github.com/cyphar/filepath-securejoin to prevent filepath parent walk attacks.